abis_mapping/templates/survey_metadata_v2/mapping.py

```diff
@@ -681,44 +682,34 @@ def add_agent(
     def add_plan(
         self,
         uri: rdflib.URIRef,
+        survey_type_attribute: rdflib.URIRef | None,
+        target_habitat_scope_attributes: Iterator[rdflib.URIRef],
+        target_taxa_attributes: Iterator[rdflib.URIRef],
         row: frictionless.Row,
         graph: rdflib.Graph,
     ) -> None:
         """Adds plan to graph.
 
         Args:
-            uri (rdflib.URIRef): Plan reference.
-            row (frictionless.Row): Raw data row.
-            graph (rdflib.Graph): Graph to be modified.
+            uri: Plan reference.
+            survey_type_attribute: SurveyType attribute for the node
+            target_habitat_scope_attribute: targetHabitatScope attribute for the node.
+            target_taxa_attribute: target taxa attribute for the node.
+            row: Raw data row.
+            graph: Graph to be modified.
         """
         # Add type
         graph.add((uri, a, rdflib.PROV.Plan))
 
         # Add attributes
-        if row["surveyType"]:
-            graph.add(
-                (
-                    uri,
-                    utils.namespaces.TERN.hasAttribute,
-                    utils.rdf.uri("attribute/surveyType", utils.namespaces.CREATEME),
-                )
-            )
-        if row["targetHabitatScope"]:
-            graph.add(
-                (
-                    uri,
-                    utils.namespaces.TERN.hasAttribute,
-                    utils.rdf.uri("attribute/targetHabitatScope", utils.namespaces.CREATEME),
-                )
-            )
-        if row["targetTaxonomicScope"]:
-            graph.add(
-                (
-                    uri,
-                    utils.namespaces.TERN.hasAttribute,
-                    utils.rdf.uri("attribute/targetTaxa", utils.namespaces.CREATEME),
-                )
-            )
+        if survey_type_attribute:
+            graph.add((uri, utils.namespaces.TERN.hasAttribute, survey_type_attribute))
+
+        for hbt_attr in target_habitat_scope_attributes:
+            graph.add((uri, utils.namespaces.TERN.hasAttribute, hbt_attr))
+
+        for tx_attr in target_taxa_attributes:
+            graph.add((uri, utils.namespaces.TERN.hasAttribute, tx_attr))
```

Comment:
I agree with removing the old code, not sure we need this new stuff.
These 3 attributes have collections I added in #287.
The attribute nodes are linked to the collections. Then the collections are linked to the `tern:Survey` node.
Not sure if we also need a link from the `prov:Plan` to the attributes themselves? Or should the `prov:Plan` link to the collections?

Reply:
no idea what any of that means. The scope for me here was to clean up a section of mapping code that was creating CREATEME IRIs incorrectly. I think the validity of the node's shape can be handled in subsequent PRs.

Reply:
oh sorry i dont see it as new. It's replacing.